x-pack/plugins/apm/common/anomaly_detection/get_preferred_service_anomaly_timeseries.ts

```diff
@@ -18,7 +18,7 @@ export function getPreferredServiceAnomalyTimeseries({
   fallbackToTransactions,
 }: {
   environment: Environment;
-  environments: Environment[];
+  environments: string[];
```

Comment:
this might be a code smell.. why is it needed? (it's useful because it signals that an environment has been validated, to not be an empty string)